### DISCUSSION TITLE
Meerk40t very slow and laggy, windows 10 64bits

### DISCUSSION THREAD

**Discussion by Badrimus** [meerk40t/meerk40t#2448] [General]
Hi, any solution regarding the lag of meerk40t? 
It runs very very slow on my laptop , I tried various releases and all the same .
Any help would be much appreciated
Cheers

**Comment by Sophist-UK** [comment]
Sorry - but we will need to know a bit more about your computer to help you with this.

Processor?
Memory?
Disk type: HDD/SSD
Operating System?

Do you get lags on any other software you run on this computer?

**Comment by Badrimus** [comment]
i7 , 16gb RAM, 1tb ssd, win 10 64bits
I don't have lags at all with any other software.
Thanks

**Comment by Sophist-UK** [comment]
@tatarize / @jpirnay ?

**Comment by tatarize** [comment]
What operation are you trying to do that is particularly slow. There's very little latency doing very basic things so there's bound to be something that is a problem operation rather than everything. For example, are you loading a file that is particular laggy. (Entirely possible and reasonable).

**Reply by Badrimus** [reply]
The software appears on the screen 15 to 20 seconds after I launched it, then for me to be able to manipulate it I have to wait Another 15 seconds, ones fully launched I can draw something for example or load a file without issue, but when I open the controller for exemple to home the laser or move it, ones I'm done and try to close the controller window it closes after 15 seconds or so.. (k40 whisperer and lightburn works fine with 0 latency)

**Reply by tatarize** [reply]
Can I maybe get a copy of your meerk40t.cfg file? 
%localappdata%/meerk40t/meerk40t.cfg

tatarize at my gmail email would be fine. This sounds like the hardware basically locks up and rather than working properly threaded it ends in disaster. So what controller are you running? And does the log if you turn on all the logging, in console `channel list` will list the channels and `channel print #` will print them out or open them to show in the regular log. My current best guess if that your controller is basically doing something I never considered and is *quickly* looping forever without a match.

You mentioned whisperer and lightburn which don't have overlapping controllers.  If you launch meerk40t with `-X` it would clear out your settings (by not loading them) and could maybe rule out if you set a weird setting that is causing rejection and somehow doesn't have any delay or something in there.

**Reply by Badrimus** [reply]
i installed meerk40t in my other laptop and it work fine my with my nano 2 board. i downloaded meerk40t.exe , it´s the only windows version right?
i will send you the file ones i connect to my other laptop that has the issue. thanks mate

**Comment by Sophist-UK** [comment]
This is not something I have heard about before, and I am guessing it is as likely to do with your operating system install as it is with MK (even though you only get lagging with MK)

My first advice is to download the ISO for your version of Windows and do an upgrade in place to refresh your copy of Windows. I know this is a time-consuming pain, but it is IMO a good first step. (This is an upgrade in place - mount the ISO and double click on the EXE file - it will not uninstall anything or lose your settings or data, just refresh all the Windows components.)

**Reply by Badrimus** [reply]
i´m using another laptop that run meerk40t sweetly, ones i have time i will try to dowload the windows iso and test it in my main laptop, for now my old laptop is doing the job 
cheers mate

**Comment by Badrimus** [comment]
I fixed the issue, what i´ve done is remove meerk40t from appdata, run the exe y voila opens up immediately. can it be duo to me have imported over 600 fonts is what was causing the lag?

**Comment by tatarize** [comment]
IIRC, you had the latest version. We do check some font stuff but it's highly optimized, though it could be entirely possible. I think I only have a hundred fonts or so. And we figured based on slow speeds we could check/parse a few thousand fonts a second. Maybe it's slower than we thought or some particular fonts could end up slowing things down more.
